scripts/generate_markdown_files.py

```diff
@@ -0,0 +1,143 @@
+                md_content += f"- **Max Length**: {api_object.max_length}\n"
+            md_content += "\n"
+
+    return md_content
+
+
+# Function to find all subclasses of Service and generate markdown files
+def generate_markdown_files(output_dir):
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    service_classes = [cls for _, cls in getmembers(class_definitions, isclass) if
+                       issubclass(cls, class_definitions.Service) and cls is not class_definitions.Service]
+
+    print(f"Detected {len(service_classes)} service classes.")
+
+    updates_made = False
+    updated_classes = []
+
+    for service_class in service_classes:
+        instance = service_class()
+        if isinstance(instance.service_name_, str) and instance.service_name_:
+            md_content = create_markdown_for_class(instance)
+            file_name = f"{service_class.__name__.lower()}.md"
+            file_path = os.path.join(output_dir, file_name)
+            if os.path.exists(file_path):
+                with open(file_path, 'r', encoding='utf-8') as md_file:
+                    current_content = md_file.read()
+                if current_content == md_content:
+                    continue
+                else:
+                    update_type = 'updated'
+            else:
+                update_type = 'created'
+            with open(file_path, 'w', encoding='utf-8') as md_file:
+                md_file.write(md_content)
+            updates_made = True
+            updated_classes.append((service_class.__name__, update_type))
+
+    return updates_made, updated_classes
+
+
+# Function to update only the 'API Reference' section in mkdocs.yml
+def update_mkdocs_yml(mkdocs_yml_path, api_dir):
+    with open(mkdocs_yml_path, 'r', encoding='utf-8') as yml_file:
+        mkdocs_config = yaml.safe_load(yml_file)
+
+    # Generate the API Reference nav entries
+    api_files = sorted(os.listdir(api_dir))
+    api_nav = [{'{}'.format(os.path.splitext(f)[0].capitalize()): os.path.join('api', f).replace("\\", "/")} for f in
+               api_files]
+
+    # Update only the 'API Reference' section in the nav
+    nav_updated = False
+    original_nav = None
+    for item in mkdocs_config['nav']:
+        if isinstance(item, dict) and 'API Reference' in item:
+            original_nav = item['API Reference']
+            if item['API Reference'] != api_nav:
+                item['API Reference'] = api_nav
+                nav_updated = True
+            break
+
+    if not nav_updated and original_nav is None:
+        mkdocs_config['nav'].append({'API Reference': api_nav})
+        nav_updated = True
+
+    if nav_updated:
+        with open(mkdocs_yml_path, 'w', encoding='utf-8') as yml_file:
+            yaml.safe_dump(mkdocs_config, yml_file, line_break='\n\n', allow_unicode=True, sort_keys=False)
+        print("Updated mkdocs.yml with API reference documentation.")
+
+    return nav_updated, original_nav, api_nav
```

Comment:
also here, `nav_updated` is not required. the `main()` script can just compare the 2 returned lists.

Reply:
Will be addressed in a separate PR.